Enchancement 92

omahonyb opened the pull request:
*Issue #, if available: #92

*Description of changes: Moving to logic.azure and new adaptive card format


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

omahonyb committed:
Adding directive for logic.azure.com

omahonyb committed:
Adding Teams Workflows adaptive card code block for AWS Orgs

omahonyb committed:
Adding accordian for ARN and Description to save screen space and pretty it up

tcicone01 commented:
Thanks, I did something similar, but yours is prettier :-)

andrewcr7 commented:
@omahonyb Thanks for the PR.  What kind of Workflow do you have this associated to on your side?

I wired it up to a "When a Teams webhook request is received" action, and that required the message be wrapped in a block like below.  So wanted to see if you set up the Workflow differently. TY!

```
{
  "type": "message",
  "attachments": [{
      "contentType": "application/vnd.microsoft.card.adaptive",
      "contentUrl": "url",
      "content": {
          YOUR CARD
      }
  }]
}
```

omahonyb commented:
@andrewcr7 
TBH I am currently moving away from workflows are they are an absolute mess! :P 
In our case we have multiple nightly flapping VPNs that are spamming the channels and the filter to ommit these just doesnt work and i decided to stop wasting time on it. 
Instead I am in the process of hooking up "AWS User Notifications". I have it working by email, just waiting for Corp to enable the App in our Teams setup.
[On another note, as I am also implementing a centralised AWS backup, I am planning on using User Notifications to send failed jobs from member accounts to the eventbridge in the management account so these will go to Teams also. unfortunately while I can monitor jobs from management we cant alert on them as that functionality doesnt exist [yet?]. this was part of my reasoning to migrate to a centralised messaging setup as implementing this in lambda or sns would be even messier]

With workflows we had to use user to post to channel, but our corp already had a bot account set up with permissions so we added that to the channel and added it as an owner to the workflow, and it all came together. Documentation is crap from MS though :P

![Screenshot 2025-03-12 at 14 12 14](https://github.com/user-attachments/assets/227d86bf-e8fa-4574-8ece-7b912a249da3)

andrewcr7 commented:
Thanks for the info on the Workflow setup.  I'm going to look at integrating a bit of this for the folks still using AHA.  

Glad you're looking at AWS User Notifications.  The improved filtering and notification aggregation features in User Notifications are helpful in reducing noise from spammy events (or remove them entirely). If you're looking for a filter for VPN events, I posted on this recently. Depending which specific event you're seeing, you can replace the eventTypeCode in the filter to trim those.
https://repost.aws/articles/AR2SoA5sd4SMa6_7HqbYOK3Q/developing-a-high-specificity-aws-health-event-strategy-with-aws-user-notifications

omahonyb commented:
Thanks for that ill have a look tomorrow.

I havent fully disabled AHA yet as I am not 100% done with User Notifications. I am having issues with getting eventbridge rules from other accounts to trigger atm [no option to choose non-default bus in UN and the default bus doesnt seem to be wanting to trigger the custom UN rule if i point the remote rule to default]

I also want to do some investigation into allowing other teams use it as we allow a fair amount of autonomy for the dev teams and id like them to be able to manage their own crap.

But even though it is relatively new, it has a lot of potential. Add to this the fact that the Teams Workflows since they depreciated proper webhooks is a disaster, the docs are rubbish, and the support forums seem to respond with "that isnt implemneted" i decided screw that :P

andrewcr7 commented:
BTW, your AWS Solutions Architect reached out a couple months back about this PR, but I lost their contact info.  If you want to separately work through User Notifications, let them know you want to set up a call and they can set that up.

omahonyb commented:
> BTW, your AWS Solutions Architect reached out a couple months back about this PR, but I lost their contact info. If you want to separately work through User Notifications, let them know you want to set up a call and they can set that up.

Erm, who? I don t remember ever speaking to one designated to us :P

I was going to post some questions on re:post but it seems my account is ballsed up. I can log in to my buildeer account, but when i try to sign into re:post i get "Your login to re:Post did not go through. This could happen due to either your AWS account access or re:Post Community manager has blocked you for violating Community Guidelines. If you have any question, please contact [AWS Support](https://aws.amazon.com/contact-us/)" and support have been zero help.

I am just back from leve so I am going to look at your advanced filters this week hopefully.

omahonyb commented:
FYI i was going to comment on your article as I seem to have an issue, but with no login to re:post i cant. I understand this isnt the correct forum for it, but step 4 "Sample Event" AWS Health Events doesnt exist anymore in the drop down, and when I create the event pattern the Test Pattern button is greyed out. And then of course if i try to sidestep this by using the send event from iinside the Bus, i cant send anything from aws.health. And it seems i cannot create a custom rule in the notification configuration with a custom service name such as test.aws.health either. Seems I am going around in circles and will have to wait for the whole thing to kick off an actual event to see if it works, which doesnt seem great :P
